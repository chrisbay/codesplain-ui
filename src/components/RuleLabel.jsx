import React, { PropTypes } from 'react';

const styles = {
  label: {
    padding: '5px',
    fontSize: '10pt',
    lineHeight: 1,
    marginTop: '5px',
    width: '100%',
    maxWidth: '200px',
    borderRadius: '5px',
    borderStyle: 'solid',
    borderWidth: '1px',
  },
}

class RuleLabel extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      isActive: false,
    }
  }

  render() {
    const { color, text, onClick } = this.props;
    const { isActive } = this.state;
    const backgroundColor = isActive ? color : 'transparent';
    const borderColor = isActive ? 'transparent' : '#e6e6e6' // light grey
    return (
      <div
        style={{...styles.label, backgroundColor, borderColor }}
        onClick={() => { 
          this.setState({ isActive: !isActive });
          onClick();
        }}
        children={text}
      />
    );
  }
}

export default RuleLabel

RuleLabel.propTypes = {
  color: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
}