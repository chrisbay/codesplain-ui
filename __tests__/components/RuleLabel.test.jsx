import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import getMuiTheme from 'material-ui/styles/getMuiTheme';

import RuleLabel from '../../src/components/RuleLabel';

describe('<RuleLabel />', () => {
  const muiTheme = getMuiTheme();
  const shallowWithContext = (node) => shallow(node, { context: { muiTheme } });

  it('matches snapshot', () => {
    const wrapper = shallowWithContext(
      <RuleLabel 
        color=""
        text=""
        onClick={jest.fn()}
      />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
  it('displays the text prop', () => {
    const text = 'Test Text';
    const wrapper = shallowWithContext(
      <RuleLabel 
        color=""
        text={text}
        onClick={jest.fn()}
      />
    );
    expect(wrapper.children()).toHaveLength(1);
    expect(wrapper.children().node).toBe(text);
  });
  it('toggles color when clicked', () => {
    const color = '#000000';
    const wrapper = shallowWithContext(
      <RuleLabel 
        color={color}
        text={''}
        onClick={jest.fn()}
      />
    );
    expect(wrapper.props().style.backgroundColor).toBe('transparent');
    wrapper.simulate('click');
    expect(wrapper.props().style.backgroundColor).toBe(color);
    wrapper.simulate('click');
    expect(wrapper.props().style.backgroundColor).toBe('transparent');
  });
  it('invokes the onClick callback prop', () => {
    const onClick = jest.fn();
    const wrapper = shallowWithContext(
      <RuleLabel 
        color={''}
        text={''}
        onClick={onClick}
      />
    );
    wrapper.simulate('click');
    expect(onClick.mock.calls.length).toBe(1);
  });
});
