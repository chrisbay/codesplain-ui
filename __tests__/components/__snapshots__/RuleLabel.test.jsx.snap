exports[`<RuleLabel /> matches snapshot 1`] = `
<div
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "transparent",
      "borderColor": "#e6e6e6",
      "borderRadius": "5px",
      "borderStyle": "solid",
      "borderWidth": "1px",
      "fontSize": "10pt",
      "lineHeight": 1,
      "marginTop": "5px",
      "maxWidth": "200px",
      "padding": "5px",
      "width": "100%",
    }
  } />
`;
